src/screens/Forum/ForumPostScreen.js

```diff
@@ -1,10 +1,182 @@
-import React from 'react';
-import { View, Text } from 'react-native';
+import React, { useState, useEffect, useCallback } from 'react';
+import {
+  View, Text, StyleSheet, RefreshControl, ActivityIndicator,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import firestore from '@react-native-firebase/firestore';
+import {
+  Avatar, Button, Card, Paragraph,
+} from 'react-native-paper';
+import { ScrollView } from 'react-native-gesture-handler';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: 'azure',
+  },
+  post: {
+    borderRadius: 15,
+    margin: 20,
+  },
+  card: {
+    borderRadius: 15,
+    marginVertical: 10,
+    marginHorizontal: 20,
+  },
+  cardBackground: {
+    borderRadius: 15,
+    paddingVertical: 15,
+    backgroundColor: 'lavender',
+  },
+});
+
+function ForumPost({
+  title, name, date, body,
+}) {
+  return (
+    <Card style={styles.post}>
+      <Card.Title title={title} subtitle={`${name} ${date}`} left={(props) => <Avatar.Text {...props} label={name.charAt(0).toUpperCase()} />} />
+      <Card.Content>
+        <Paragraph>{body}</Paragraph>
+      </Card.Content>
+      <Card.Actions>
+        <Button>Reply</Button>
+      </Card.Actions>
+    </Card>
+  );
+}
+
+function ForumReply({
+  name, date, body,
+}) {
+  return (
+    <Card style={styles.card}>
+      <Card.Title subtitle={`${name} ${date}`} />
+      <Card.Content>
+        <Paragraph>{body}</Paragraph>
+      </Card.Content>
+    </Card>
+  );
+}
+
+export default function ForumPostScreen({ navigation }) {
+  const postId = navigation.getParam('postId');
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [post, setPost] = useState(null);
+  const [replies, setReplies] = useState([]);
+  const [refreshing, setRefreshing] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const getData = useCallback(async () => {
```

Comment:
Hey sorry about my last comment but I made a mistake. We don't want to put `async` here; instead, we want `getData()` to return a promise that gets resolved only after all the firestore calls are resolved. This means we need to add some returns to this function. See below.